core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -1302,7 +1302,7 @@ class Partition(val topicPartition: TopicPartition,
   }
 
   def appendRecordsToLeader(records: MemoryRecords, origin: AppendOrigin, requiredAcks: Int,
-                            requestLocal: RequestLocal, verificationGuard: VerificationGuard = VerificationGuard.SENTINEL): LogAppendInfo = {
+                            bufferSupplier: BufferSupplier, verificationGuard: VerificationGuard = VerificationGuard.SENTINEL): LogAppendInfo = {
```

Comment:
Hmm, why are we changing classes that we don't intend to move to the storage module?